client-misk/src/main/kotlin/app/cash/backfila/client/misk/DeleteBy.kt

```diff
@@ -0,0 +1,20 @@
+package app.cash.backfila.client.misk
+
+import java.time.Instant
+import java.time.LocalDate
+import java.time.ZoneOffset.UTC
+
+/**
+ * Annotation used on your backfill class to indicate that this backfill should continue to be
+ * useful until the specified date. Most backfills should not specify anything which defaults to as
+ * soon as possible. No removal reminders will occur until after this date.
+ *
+ * deleteBy : Must have a format of YYYY-MM-DD.
+ */
+@Target(AnnotationTarget.VALUE_PARAMETER, AnnotationTarget.CLASS)
+@Retention(AnnotationRetention.RUNTIME)
+annotation class DeleteBy(val deleteBy: String)
+
+fun String.parseDeleteByDate(): Instant {
```

Comment:
We are using only Java time dependencies here. Do you think this should go somewhere else? I think it is handy to see this underlying implementation.

Reply:
I guess it can be here but you should use `internal`. Global extensions infect every user of the library. I also don't personally like extending String since this is not applicable to any String, just your particular String

Reply:
I see its convenient to have the extension in files above, seems fine I would just suggest `internal`

Reply:
why not just make this a function or extension on `DeleteBy`

Reply:
Yeah it is much better as an extension on DeleteBy